src/compiler/scala/tools/nsc/backend/jvm/BCodeHelpers.scala

```diff
@@ -841,6 +841,17 @@ abstract class BCodeHelpers extends BCodeIdiomatic with BytecodeWriters {
       }
     }
 
+    /*
+     * must-single-thread
+     */
+    def emitParamNames(jmethod: asm.MethodVisitor, params: List[ValDef]) = {
+      for (param <- params) {
+        var access = asm.Opcodes.ACC_FINAL
+        if (param.mods.isArtifact)
+          access |= asm.Opcodes.ACC_SYNTHETIC
```

Comment:
It seems odd to me that you're looking up flags and the name from `ValDef#{name, mods}`, rather than from getting then from the `Symbol`. Usually, after typechecking, the `Symbol` is the used as the source of truth.